include/range/v3/utility/iterator.hpp

```diff
@@ -480,25 +523,111 @@ namespace ranges
                 return *this;
             }
         };
+
+        namespace detail
+        {
+            template<typename I>
+            struct reverse_cursor
+            {
+            private:
+                template<typename OtherI>
+                friend struct reverse_cursor;
+                I it_;
+            public:
+
+                struct mixin : basic_mixin<reverse_cursor>
+                {
+                    RANGES_CXX14_CONSTEXPR mixin() = default;
+                    RANGES_CXX14_CONSTEXPR mixin(reverse_cursor pos)
+                        : basic_mixin<reverse_cursor>{std::move(pos)}
+                    {}
+                    RANGES_CXX14_CONSTEXPR mixin(I it) : mixin(reverse_cursor{it})
+                    {}
+                    RANGES_CXX14_CONSTEXPR I base() const
+                    {
+                        return this->get().base();
+                    }
+                };
+
+                RANGES_CXX14_CONSTEXPR reverse_cursor(I it) : it_(std::move(it)) {}
+                template<typename U, CONCEPT_REQUIRES_(Convertible<U, I>{})>
+                RANGES_CXX14_CONSTEXPR reverse_cursor(reverse_cursor<U> const& u)
+                        : it_(u.base()) {}
+
+                template<typename U, CONCEPT_REQUIRES_(Convertible<I, U>{})>
+                RANGES_CXX14_CONSTEXPR operator reverse_cursor<U>() {
+                    return U{it_};
+                }
+
+                RANGES_CXX14_CONSTEXPR reverse_cursor() = default;
+
+                RANGES_CXX14_CONSTEXPR
+                auto current() const -> decltype(*it_)
+                {
+                    I tmp(it_);
+                    return *(--tmp);
+                }
+                RANGES_CXX14_CONSTEXPR
+                I base() const
+                {
+                    return it_;
+                }
+                RANGES_CXX14_CONSTEXPR
+                bool equal(reverse_cursor const& that) const
+                {
+                    return it_ == that.it_;
+                }
+                RANGES_CXX14_CONSTEXPR
+                void next()
+                {
+                    --it_;
+                }
+                RANGES_CXX14_CONSTEXPR
+                void prev()
+                {
+                    ++it_;
+                }
+                CONCEPT_REQUIRES(RandomAccessIterator<I>())
+                RANGES_CXX14_CONSTEXPR
+                void advance(iterator_difference_t<I> n)
+                {
+                    it_ -= n;
+                }
+                template<class U, CONCEPT_REQUIRES_(RandomAccessIterator<I>())>
+                RANGES_CXX14_CONSTEXPR
+                iterator_difference_t<I>
+                distance_to(reverse_cursor<U> const &that) const
+                {
+                    return it_ - that.base();
+                }
```

Comment:
This doesn't need to be a template. The machinery in `basic_iterator` makes sure the comparison is symmetric by forcing a conversion.